### PR TITLE
[TK-05217] add authored prefix

### PR DESCRIPTION
**Pull request by freesig**
- Adds an AuthoredPrefix
- Change SourceChain to use the AuthoredPrefix
- Adds authored stores to the required workspaces
- Adds authored stores to the cascade 
  - This also includes combining the cache with the authored store as hash sets
- Removes the source chain cache constructor
- Changes integrate_to_cache to take integrate_to_authored

Also added some helpers for writing conductor tests. It's becoming very easy to write these tests now which is nice.

### CODE REVIEW

**Comment by freesig** [comment]
Actually there's a flaw in this. The authored and cache need to be recombined to get the "latest view". I thought I could just chuck them into hashsets but there's a bit more too it then that.
We really need to combine them into a scratch somehow.
I'm going to put a little more thought into this.
One thing we could do is "fetch" from the authored into the cache but then we end up always storing the data in authored and cache.
What we really need is an easy way to take the Union of two metadata stores.

**Comment by freesig** [comment]
I have added some changes that make this correct. I think there's further thought that needs to go into computing values on write because the way our system works means we are often writing as we are reading so we can accumulate the most recent view. This results in "compute on writes" becoming "compute on read". This was already the case but I think it's become more obvious in this PR

File: crates/holochain/src/core/state/cascade.rs [hunk 1]
```diff
@@ -678,22 +763,55 @@ where
...
+
+        let cache_data = ok_or_return!(self.cache_data.as_ref(), None);
+        let authored_data = ok_or_return!(self.authored_data.as_ref(), None);
+        // Meta Cache
+        let oldest_live_element =
+            match self.get_oldest_live_element(&entry_hash, &DbPair::from(cache_data))? {
+                // Look for the element data in authored
+                Some(Search::Continue(oldest_live_header)) => {
+                    match authored_data.element.get_element(&oldest_live_header)? {
+                        Some(element) => Search::Found(element),
```
Comment (neonphog) [inline crates/holochain/src/core/state/cascade.rs:805]: I'm not sure about this logic. If you search the cache first, and it is mostly clear, but has a much newer live element, won't it return that, even though the authored_data might have an older live element?

**Review by neonphog — APPROVED** [review]
Not sure about the one logic comment - but approving - code looks good, and I'm sure there are hundreds of corner cases we'll have to regression test eventually : )

File: crates/holochain/src/core/state/cascade.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain/src/core/state/cascade.rs:805]: hmm good catch, seems I need to actually gather all the elements together to search from both stores
